DOC-2267: Add new documentation for `ImportWord` Plugin. (#3094)
tinymce/tinymce-docs: modules/ROOT/pages/importword.adoc

= Import from Word plugin
:navtitle: Import from Word
:description: Provides a way to import .docx (Word documents) or .dotx (Word templates) files into the editor.
:description_short: Import .docx or .dotx files into the editor.
:keywords: plugin, importword, import from word
:pluginname: Import from Word
:plugincode: importword
:plugincategory: premium

> include::partial$misc/admon-paid-addon-pricing.adoc[]

Reviewer: @kemister85 I think we need something like this, which I picked up from AI assistant docs.

`NOTE: This plugin is only available as a link:{aipricingurl}/[paid add-on] to link:{pricingpage}/[a TinyMCE subscription].`

Author: Updating this as part of the last commit, resolving this message.

include::partial$misc/admon-requires-7.0v.adoc[]

The {pluginname} plugin lets you import `.docx` (Word document) or `.dotx` (Word template) files into the editor. The process preserves formatting and rich media.

== Interactive example

[NOTE]
.{pluginname} demo
====

> Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/importword/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.

Author: ```suggestion
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/import-from-word/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.
```

> Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/import-from-word/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.

Reviewer: ```suggestion
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/import-from-word/[the {pluginname} demo], to try it out today.
```

Reviewer: @kemister85 Let me confirm if import from word is added to home page demo.

Author: As discussed, we will update this link in a seperate ticket, so we can commence QA for the rest of DOCs 7.0

====

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'code importword',
  toolbar: 'importword code',
  importword_service_url: '<service_URL>'
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

> include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

Reviewer 2: ```suggestion
include::partial$configuration/{plugincode}.adoc[leveloffset=+1]
```
fix nav bar levels

= Import from Word plugin
:navtitle: Import from Word
:description: Provides a way to import .docx (Word documents) or .dotx (Word templates) files into the editor.
:description_short: Import .docx or .dotx files into the editor.
:keywords: plugin, importword, import from word
:pluginname: Import from Word

> // values for <tier>
> // opensource plugin = tieroneplan
> // premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan
> :pluginminimumplan: <tier>

Author: ```suggestion
```

= Import from Word plugin
:navtitle: Import from Word
:description: Provides a way to import .docx (Word documents) or .dotx (Word templates) files into the editor.
:description_short: Import .docx or .dotx files into the editor.
:keywords: plugin, importword
:pluginname: Import from Word
// values for <tier>
// opensource plugin = tieroneplan
// premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan
:pluginminimumplan: <tier>
:plugincode: importword
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

The {pluginname} plugin lets you import `.docx` (Word document) or `.dotx` (Word template) files into the editor. The process preserves formatting and rich media.

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'code importword',
  toolbar: 'importword code',

>   importword_service_url: '<replace_with_service_url>'

Author: ```suggestion
  importword_service_url: '<service_URL>'
```

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'code importword',
  toolbar: 'importword code',
  importword_service_url: '<replace_with_service_url>'
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]

include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[]

// Remove is not required before merging
////
== Events

The {pluginname} plugin provides the following events.

include::partial$events/{plugincode}-events.adoc[]

== APIs

The {pluginname} plugin provides the following APIs.

include::partial$plugin-apis/{plugincode}-apis.adoc[]

> ////

Reviewer 3: We do have the `ImportWord` command which can be used, but no events.
The `ImportWord` command needs to be called from a true user interaction, for example in the callback of a button click event, but it won't work from console

> ////

Author: I think we can leave this out, if you agree? I have already removed the placeholder for this, but happy to add it back if you think we need it.
